obspy/core/util/base.py

```diff
@@ -60,6 +59,30 @@
 NATIVE_BYTEORDER = _sys_is_le and '<' or '>'
 
 
+# apply simple memoization cache on load_entry_points
+# this function cannot go in decorator.py due to circular import issues
+def _load_entry_point_decorator(func):
+    """
+    Decorate pkg_resources' load_entry_point function to cache outputs.
+    """
+    cache = {}
+    expected_keys = ('dist', 'group', 'name')
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # make a str key from the inputs to func for hashing in cache
+        input_dict = inspect.getcallargs(func, *args, **kwargs)
+        hash_str = '/'.join([input_dict[x] for x in expected_keys])
+        if hash_str not in cache:
+            cache[hash_str] = func(*args, **kwargs)
+        return cache[hash_str]
+
+    return wrapper
+
+
+load_entry_point = _load_entry_point_decorator(load_entry_point)
```

Comment:
> Yes this will only affect obspy.core.util.base.load_entry_point so all should be good.

Still a bit confused about this, as we import `from pkg_resources` throughout the code base wherever this function is used and not `from obspy.core.util.base`..

```
obspy/core/event/catalog.py
32:from pkg_resources import load_entry_point
496:            write_format = load_entry_point(

obspy/core/inventory/inventory.py
20:from pkg_resources import load_entry_point
270:            write_format = load_entry_point(

obspy/core/util/base.py
28:from pkg_resources import iter_entry_points, load_entry_point
62:# apply simple memoization cache on load_entry_points
64:def _load_entry_point_decorator(func):
66:    Decorate pkg_resources' load_entry_point function to cache outputs.
73:        print("in _load_entry_point_decorator")
84:load_entry_point = _load_entry_point_decorator(load_entry_point)
347:    func = load_entry_point(entry_point.dist.key, 'obspy.plugin.%s' % (group),
396:            is_format = load_entry_point(
426:        read_format = load_entry_point(
495:        func = load_entry_point(ep.dist.key,

obspy/core/stream.py
25:from pkg_resources import load_entry_point
1436:            write_format = load_entry_point(

obspy/core/tests/test_waveform_plugins.py
14:from pkg_resources import load_entry_point
49:                is_format = load_entry_point(
205:            is_format = load_entry_point(
462:            fct_name = module_name + "." + load_entry_point(
```

But at least in an interactive IPython session I don't see this affecting `pkg_resources`, so it indeed seems to be fine, I guess.